### ISSUE TITLE
Overall logic of the app needs to be consolidated a bit.

### ISSUE THREAD

**Issue by euroclydon37** [CoryWritesCode/tictactoe_shipt#2]
Just some constructive criticism. Don't take this to mean you didn't do a great job. You did really well.

The main issue I see with this app is that the board logic responsibility is shared across components.

You have an array of board positions in `constants.js`, but then the `GameBoard` has the array of `xSpaces` and the array of `oSpaces`. There's also a an array of winning combinations. But `GameBoard` doesn't use this. The `Box` component does.

What I would have loved to see is a reducer function (`useReducer` is awesome) that returns a 2 dimensional array full of `null`s, `'X'`s and `'O'`s. And `onPress` in the `Box` components could simply be action dispatches. The reducer can place the value at the appropriate place, update the turn, check for winners, etc.

TL;DR, the game logic and state needs to live in one place. Single source of truth.

Again, you did great. I can tell you really understand what react is doing and how to use it.
